crates/boundless-market/src/client.rs

```diff
@@ -570,12 +580,14 @@ impl Client<Http<HttpClient>, ProviderWallet, BuiltinStorageProvider> {
         set_verifier_address: Address,
         order_stream_url: Option<Url>,
         storage_provider: Option<BuiltinStorageProvider>,
+        balance_alerts: Option<BalanceAlertConfig>,
```

Comment:
Opinion: maybe this should be done as a builder pattern to avoid breaking changes? I feel like this `from_parts` public method is already unweildy. Also possible we can just refactor and address in the future, just noting.